src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const innerSeparator = separator + '';
```

Comment:
зараз в тебе не явна зміна тупу, краще використовувати `String` для приведення значення в строку
```suggestion
    const innerSeparator = String(separator);
```